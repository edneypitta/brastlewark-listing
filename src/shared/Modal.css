.backdrop {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: rgba(0,0,0,0.3);
  padding: 50px;
}

.backdrop .modal {
  background-color: #fff;
  border-radius: 5px;
  max-width: 50vw;
  margin: 0 auto;
  padding: 30px;
  color: #000;
}

.backdrop .modal .footer  {
  text-align: right;
}