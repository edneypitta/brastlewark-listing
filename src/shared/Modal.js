import React from 'react'
import PropTypes from 'prop-types'
import './Modal.css'

const Modal = props => {
  if (!props.show) {
    return null
  }

  return (
    <div className="backdrop">
      <div className="modal">
        {props.children}
        <div className="footer">
          <button onClick={props.onClose}>
            Close
          </button>
        </div>
      </div>
    </div>
  )
}

Modal.propTypes = {
  onClose: PropTypes.func.isRequired,
  children: PropTypes.element
}

export default Modal