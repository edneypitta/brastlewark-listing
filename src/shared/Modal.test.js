import React from 'react'
import { shallow } from 'enzyme'
import Modal from './Modal'

describe('Modal', () => {
  it('renders nothing if show is false', () => {
    const component = shallow(<Modal show={false} />)
    expect(component).toMatchSnapshot()
  })

  it('renders if show is true', () => {
    const component = shallow(<Modal show={true} />)
    expect(component).toMatchSnapshot()
  })
})
