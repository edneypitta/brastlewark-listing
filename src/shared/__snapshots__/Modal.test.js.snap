// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders if show is true 1`] = `
<div
  className="backdrop"
>
  <div
    className="modal"
  >
    <div
      className="footer"
    >
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`Modal renders nothing if show is false 1`] = `""`;
